Makefile

```diff
@@ -317,6 +313,48 @@ else ifeq ($(AUTO_TEST), vsock)
 		|| (echo "Vsock test failed" && exit 1)
 endif
 
+ASTERINAS_DISK ?= $(abspath target/nixos/asterinas.img)
+
+# Build the Asterinas NixOS ISO installer image
+.PHONY: iso
+iso: BOOT_PROTOCOL = linux-efi-handover64
+iso:
+	@make kernel
+	@nix-build \
+		distro/iso-image.nix \
+		--arg distro $(DISTRO_PATH) \
+		--arg kernel $(KERNEL_PATH) \
+		--arg tools $(NIXOS_TOOLS_PATH) \
+		--arg autoInstall $(AUTO_INSTALL) \
+		--out-link target/nixos/iso_image
+
+# Build the Asterinas NixOS ISO installer image and then do installation
+.PHONY: run_iso
+run_iso: OVMF = off
+run_iso: iso
+	@./tools/nixos/run_iso.sh target/nixos
+
+# Create an Asterinas NixOS installation on host
+.PHONY: nixos
+nixos: BOOT_PROTOCOL = linux-efi-handover64
+nixos:
+	@make kernel
```

Comment:
Likewise, the `kernel` Makefile target should be added to the `nixos` target as a dependency.